src/hooks/firestore-hooks.test.ts

```diff
@@ -34,7 +34,7 @@ var mockOnSnapshot = (callback: (snap: any) => void) => {
   }));
 };
 var mockDelete = jest.fn();
-var mockDocRef = jest.fn((path: string) => ({
+var mockDoc = jest.fn((path: string) => ({
```

Comment:
Doesn't the test environment/runner support modern (actually several years old) JS features, like `let` ?

Reply:
The answer to that is more complicated than you might imagine. For mocking purposes, jest hoists `jest.mock()` calls to the top of the module. For historical reasons, `var` is hoisted to the top of the block in which it's declared by the JS engine. `let` and `const`, however, are not hoisted, which means that if you try to access an identifier declared with `let` or `const` from within your `jest.mock()` implementation, you get an `identifier not defined` error. Therefore, our lint rules (which otherwise warn/error on use of `var`) have an exception for `.test.*` files to allow `var` and I routinely use `var` when defining variables used in `jest.mock()` implementations due to the hoisting semantics.

Reply:
Thanks for the clarification. It makes more sense now, even if unfortunate, IMO.